rotations/demonhunter/vengeance.lua

```diff
@@ -61,42 +58,33 @@ local _Ranged = {
 
 local _Melee = {
 	-- Rotation
-	{{ -- infront
+	{{ -- In front
 		{ "Fiery Brand", { "!player.buff(Demon Spikes)", "!player.buff(Metamorphosis)" }},
-
 		{{
-			{ "Demon Spikes", "player.spell(Demon Spikes).charges >= 2" },
-			{ "Demon Spikes", "!player.buff(Demon Spikes)" },
+			{ "Demon Spikes", "player.spell(Demon Spikes).charges = 2" },
+			{ "Demon Spikes", { "!player.buff(Demon Spikes)", "player.health < 85" }}, -- This will prevent DS to be used unnecessarily on CD and possibly save 1 charge for better manual usage
 		}, { "!target.debuff(Fiery Brand)", "!player.buff(Metamorphosis)" }},
-
-		{ "Empower Wards", "target.casting.time < 2" },
-
-		{ "Spirit Bomb", "!target.debuff(Frailty)" },
-
+		--{ "Empower Wards", "target.casting.time < 2" }, -- This isn't working atm
+		{ "Spirit Bomb", { "talent(6,3)", "!target.debuff(Frailty)", "player.buff(Soul Fragments).count >= 1" }}, -- It will only attempt to cast Spirit Bomb if there is at least 1 Soul Fragment around
 		{ "Soul Carver", "target.debuff(Fiery Brand)" },
-
 		{ "Immolation Aura", "player.pain <= 80" },
-
-		{ "Felblade", "player.pain <= 70" },
-
-		{ "Soul Barrier" },
-
-		{ "Soul Cleave", "player.buff(Soul Fragments).count >= 5" }, -- <-- Don't use this automatically
-
-		{ "Fel Erruption" },
-
+		{ "Felblade", { "talent(3,1)", "player.pain <= 70" }},
+		{ "Soul Barrier", "talent(7,3)" },
+		{ "Fracture", { "talent(4,2)", "player.pain >= 60", "player.buff(Soul Fragments).count > 5" }}, -- This will make this talent more usable
+		{ "Soul Cleave", "player.buff(Soul Fragments).count >= 5" },
 		{ "Soul Cleave", "player.pain >= 80" },
-
 		{ "Shear" },
 	}, 'target.infront' },
 }
 
 local _AoE = {
+	{ "Immolation Aura", "player.pain <= 80" },
+	{ "Sigil of Flame", "talent(6,1)" },
 }
 
 NeP.Engine.registerRotation(581, '[|cff'..NeP.Interface.addonColor..'NoC|r] Demon Hunter - Vengeance',
 	{ -- In-Combat
-		{ '%pause', 'keybind(shift)'},
+		--{ '%pause', 'keybind(shift)'}, -- I don't think this is necessary
```

Comment:
I think this is still necessary if you want to tie pause to the shift keybind.  There is nothing in the core NeP engine that currently provides this.